### PR TITLE
New Catalan Cheat Sheet

### PR DESCRIPTION
**Pull request by rubikkk**
###### What does your Pull Request do (check all that apply)?

Choose the most relevant items and use the following title template to name
your Pull Request.

- [X] New Instant Answer
    - [X] Cheat Sheets: **`New {Cheat Sheet Name} Cheat Sheet`**
    - [ ] Other: **`New {IA Name} Instant Answer`**
- [ ] Improvement
    - [ ] Bug fix: **`{IA Name}: Fix {FIX typos}`**
    - [ ] Enhancement: **`{IA Name}: {Adding Food and drink}`**
- [ ] Non–Instant Answer
    - [ ] Other (Role, Template, Test, Documentation, etc.): **`{GoodieRole/Templates/Tests/Docs}: {Short Description}`**

###### Description of changes

A cheat sheet for the Catalan language.

###### People to notify (@mention interested parties)

---

Instant Answer Page: https://duck.co/ia/view/catalan_cheat_sheet

[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @mention

### CODE REVIEW

**Comment by daxtheduck** [comment]
@rubikkk Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please head to https://quackslack.herokuapp.com/ to get an invite.

>More Info: http://docs.duckduckhack.com/submitting/submitting-overview.html

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Catalan Cheat Sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/catalan_cheat_sheet_mobile.png)

**Comment by tagawa** [comment]
Hi @rubikkk, thanks very much for submitting this. As you may have seen, it's been installed to the beta server for testing: https://beta.duckduckgo.com/?q=catalan+cheat+sheet&ia=cheatsheet
We'll ask for other Catalan speakers to review it first before merging.

**Comment by tagawa** [comment]
@rubikkk We have some feedback from Twitter - please could the upside-down exclamation marks and question marks be removed from the start of phrases? Apparently they should only appear at the end in Catalan, unlike Spanish: https://twitter.com/Xaviju/status/753570372906262528

**Comment by Xaviju** [comment]
There is also a small typo:
```¿On estÀ el lavabo?`` should be ```On està el lavabo?```

**Comment by rubikkk** [comment]
completely agree, I correct the typos and I also add a few more sentences
and I recommit the code.

thanks!

david.

**Comment by mcornella** [comment]
The `where is` forms are translated to "_on **és**_" (ON EHS) in Catalan. There is a little bit of confusion between when to use `és` (verb `ser`) vs. `està` (verb `estar`); in this case it's clear though: when talking about location you should use the verb `ser`. Some dialects of Catalan do use `estar` in all cases though, like Valencian.

So the following examples should be:
```
Where is the bathroom?
On és el lavabo?
ON EHS el la-BA-boo
```
```
Where is my room?
On és la meva habitació?
ON EHS la me-bah ah-bee-ta-SIOH
```

`EHS` is pronounced like the spelling of the letter `s`.

The usage of `ser` / `estar` is complicated though, and it's a big source of confusion even between native speakers. Hope that helps!

**Comment by Xaviju** [comment]
Valencians we say ```estar``` but is recommended only for colloquial use. Both are correct.
Reference: chapter 33 Gramática normativa valenciana. Link to PDF: http://www.avl.gva.es/dms/GNV

**Comment by mcornella** [comment]
Thanks! I read that it was more like a norm in Valencian, so the clarification helps.

**Comment by rubikkk** [comment]
I corrected everything and added some bonus tracks ;)

commiting!

d.

**Comment by rubikkk** [comment]
i actually did it now! :)

**Comment by rubikkk** [comment]
parsing errors, sorry no sleep :( corrected.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Catalan Cheat Sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/catalan_cheat_sheet_mobile.png?AN8teDDleQ)

**Comment by rubikkk** [comment]
when I duckpan test I get this error:


README.md
catalan.json
Terminal
 ×
spanish.json
 ×


# Testing catalan.................. 1 FAILURE                                                                                                                                  
#        -> FAIL: invalid JSON                                                                                                                                                 
                                                                                                                                                                               
    #   Failed test 'FAIL: invalid JSON'                                                                                                                                       
    #   at t/CheatSheets/CheatSheetsJSON.t line 274.                                                                                                                           
    # Looks like you failed 1 test of 1.                                                                                                                                       
                                                                                                                                                                               
#   Failed test 'can_build'                                                                                                                                                    
#   at t/CheatSheets/CheatSheetsJSON.t line 275. 

and I don't know how to get it corrected.

but anyway, I uploaded a new version of the code with the fixes and extra fields, and looks like you didn't get to see it, i will commit again and see..

File: share/goodie/cheat_sheets/json/language/catalan.json [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+		}, {
+			"key": "Can I enter/come in?",
+			"val": "Puc passar?",
+			"trn": "Pook pah-SAH"
+		}, {
+			"key": "Where is my room?",
+			"val": "On és la meva habitació?",
+			"trn": "ON ehs lah me-bah ah-bee-tah-SIOH"
+		}],
+		"Food and drink": [{
```
Comment (ManrajGrover) [inline share/goodie/cheat_sheets/json/language/catalan.json]: This should be `Food and Drink` (Notice the upper-cased D). It should match with what is mentioned in `section_order`.

**Comment by ManrajGrover** [comment]
@rubikkk I've left a comment. Try fixing it. That should make the build pass. :smile:

**Comment by rubikkk** [comment]
@ManrajGrover oh I didn't notice it! Thanks, fixing it in 3, 2, 1...! :)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Catalan Cheat Sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/catalan_cheat_sheet_mobile.png?9WGHu24sHA)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Catalan Cheat Sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/catalan_cheat_sheet_mobile.png?nK7nH41WvR)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Catalan Cheat Sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/catalan_cheat_sheet_mobile.png?6HjHaJf61I)

File: share/goodie/cheat_sheets/json/language/catalan.json [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+				"key": "Please",
+				"val": "Si us plau.",
+				"trn": "SEE OOS-plaoo"
+			}, {
+				"key": "Thank you.",
+				"val": "Gràcies.",
+				"trn": "GRA-see-əhs"
+			},
+
+			{
```
Comment (tagawa) [inline share/goodie/cheat_sheets/json/language/catalan.json]: Please could you remove this empty line here?

File: share/goodie/cheat_sheets/json/language/catalan.json [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+				"val": "Benvingut!",
+				"trn": "behn-been-GOOT"
+			}, {
+				"key": "Goodbye.",
+				"val": "Adéu.",
+				"trn": "ah-DEhoo"
+			}, {
+				"key": "Good Morning.",
+				"val": "Bon dia.",
+				"trn": "ah-DEhoo"
```
Comment (tagawa) [inline share/goodie/cheat_sheets/json/language/catalan.json]: This should be "bon dee-ah"

File: share/goodie/cheat_sheets/json/language/catalan.json [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+			}, {
+				"key": "Good Evening.",
+				"val": "Bona tarda.",
+				"trn": "ah-DEhoo"
```
Comment (tagawa) [inline share/goodie/cheat_sheets/json/language/catalan.json]: This should be "bon-ah tar-dah"

File: share/goodie/cheat_sheets/json/language/catalan.json [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+			}, {
+				"key": "Good Night.",
+				"val": "Bona nit.",
+				"trn": "ah-DEhoo"
```
Comment (tagawa) [inline share/goodie/cheat_sheets/json/language/catalan.json]: This should be "bon-ah neet"

File: share/goodie/cheat_sheets/json/language/catalan.json [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+		}, {
+			"key": "Water",
+			"val": "Aigua",
+			"trn": "I-guah"
+		}, {
+			"key": "Wine",
+			"val": "Vi",
+			"trn": "Bee"
+		}, {
+			"key": "Bread with tomatoe",
```
Comment (tagawa) [inline share/goodie/cheat_sheets/json/language/catalan.json]: Small typo: "tomatoe" should be "tomato"

File: share/goodie/cheat_sheets/json/language/catalan.json [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+			"val": "Com et dius?",
+			"trn": "Kom ət deeoos"
+		}, {
+			"key": "What time is it?",
+			"val": "Quina hora és?",
+			"trn": "Kee-noh-REHS"
+		}, {
+			"key": "Can I enter/come in?",
+			"val": "Puc passar?",
+			"trn": "Pook pah-SAH"
```
Comment (tagawa) [inline share/goodie/cheat_sheets/json/language/catalan.json]: Please change to:
`"val": "Que puc passar?",`
`"trn": "Kuh pook pah-SAH"`

**Comment by tagawa** [comment]
@rubikkk I've added a few more comments from people on Twitter. Also, we've had these two suggested phrases. Please could you add them?

You're welcome => De res.
Prounced: Dah RAYS

How much does this cost? -> Quant costa això?
Pronounced: Koo-and cost-ah asho

File: share/goodie/cheat_sheets/json/language/catalan.json (hunk 7 above)
Comment (Xaviju) [inline share/goodie/cheat_sheets/json/language/catalan.json]: I think is correct as it is, without ```Que```. 
This is a regional dialect expression, not common everywhere and not necessary to understand the sentence.

Correct
 ```
"key": "Can I enter/come in?",
"val": "Puc passar?",
"trn": "Pook pah-SAH"
```

Comment (tagawa) [inline share/goodie/cheat_sheets/json/language/catalan.json]: Thanks for the explanation. I'll remove my original comment.

File: share/goodie/cheat_sheets/json/language/catalan.json (hunk 4 above)
Comment (mcornella) [inline share/goodie/cheat_sheets/json/language/catalan.json]: Also, "evening" is `vespre`, you should use "Good Afternoon" instead.

File: share/goodie/cheat_sheets/json/language/catalan.json (hunk 7 above)
Comment (rubikkk) [inline share/goodie/cheat_sheets/json/language/catalan.json]: thanks everybody for the comments, all updated and commited!

d.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Catalan Cheat Sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/catalan_cheat_sheet_mobile.png?jpwylwisne)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Catalan Cheat Sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/catalan_cheat_sheet_mobile.png?IzWfUHtLkN)

**Comment by tagawa** [comment]
Thank you @rubikkk! It looks good on the beta server: https://beta.duckduckgo.com/?q=Catalan+Cheat+Sheet&ia=cheatsheet
so let's give it a couple more days for final feedback and then we should be good to merge.

File: share/goodie/cheat_sheets/json/language/catalan.json [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+		}, {
+			"key": "Water",
+			"val": "Aigua",
+			"trn": "I-guah"
+		}, {
+			"key": "Wine",
+			"val": "Vi",
+			"trn": "Bee"
+		}, {
+			"key": "Bread with tomatofafa",
```
Comment (mcornella) [inline share/goodie/cheat_sheets/json/language/catalan.json]: Typo on `tomato`,

File: share/goodie/cheat_sheets/json/language/catalan.json [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+				"trn": "bon-ah TAR-dah"
+			}, {
+				"key": "Good Night.",
+				"val": "Bona nit.",
+				"trn": "bon-ah NEET"
+			}, {
+				"key": "See you later.",
+				"val": "Fins després.",
+				"trn": "FEENS-dehs-prehs"
+			}
```
Comment (mcornella) [inline share/goodie/cheat_sheets/json/language/catalan.json]: This section (basics) is differently indented than the other sections.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Catalan Cheat Sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/catalan_cheat_sheet_mobile.png?Vv8OnNFveu)

File: share/goodie/cheat_sheets/json/language/catalan.json (hunk 9 above)
Comment (rubikkk) [inline share/goodie/cheat_sheets/json/language/catalan.json]: corrected typo and idention!

d.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Catalan Cheat Sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/catalan_cheat_sheet_mobile.png?KUZ7iGbHL6)

**Comment by tagawa** [comment]
Thank you reviewers for the very helpful comments, and to @rubikkk for implementing them all. 
After all the feedback, it looks like everything's good on the beta server so let's merge this. It should go live in a few days.
Thanks again everyone! 🎉

**Pull request merged**